File: clients/csharp/Exceptions/CollectionNotFoundException.cs
namespace LimyeDB.Exceptions;

/// <summary>
/// Thrown when the requested collection does not exist.
/// </summary>
public class CollectionNotFoundException : LimyeDBException
{
    /// <summary>
    /// The name of the collection that was not found.
    /// </summary>
    public string CollectionName { get; }

    public CollectionNotFoundException(string collectionName)
        : base($"Collection not found: {collectionName}", 404)
    {
        CollectionName = collectionName;
    }

    public CollectionNotFoundException(string collectionName, string message)
        : base(message, 404)
    {
        CollectionName = collectionName;
    }
}

File: clients/java/src/main/java/com/limyedb/exceptions/CollectionNotFoundException.java
package com.limyedb.exceptions;

/**
 * Thrown when the requested collection does not exist.
 */
public class CollectionNotFoundException extends LimyeDBException {
    private final String collectionName;

    public CollectionNotFoundException(String collectionName) {
        super("Collection not found: " + collectionName, 404);
        this.collectionName = collectionName;
    }

    public CollectionNotFoundException(String collectionName, String message) {
        super(message, 404);
        this.collectionName = collectionName;
    }

    /**
     * Returns the name of the collection that was not found.
     */
    public String getCollectionName() {
        return collectionName;
    }
}
